### PR TITLE
more work on suspend, also handle resume

### PR DESCRIPTION
**Pull request by thillux**
* resume handler
* fixes for NixOS
* remove shell dependency

### CODE REVIEW

File: frontends/server-signal-helper/server_signal_helper.c [hunk 1]
```diff
@@ -0,0 +1,138 @@
+#include <getopt.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <signal.h>
+#include <errno.h>
+#include <esdm_rpc_client.h>
+
+bool signal_suspend(char *pidfile_path)
```
Comment (smuellerDD) [inline frontends/server-signal-helper/server_signal_helper.c:9]: Please make it static

File: frontends/server-signal-helper/server_signal_helper.c [hunk 2]
```diff
@@ -0,0 +1,138 @@
+#include <getopt.h>
```
Comment (smuellerDD) [inline frontends/server-signal-helper/server_signal_helper.c:1]: Please add header as usual.

File: frontends/server-signal-helper/server_signal_helper.c [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+	unsigned char dummy_data[1] = { 0 };
+
+	esdm_invoke(esdm_rpcc_write_data(dummy_data, sizeof(dummy_data)));
+
+	esdm_rpcc_fini_unpriv_service();
+
+	return ret == 0;
+}
+
+void usage()
```
Comment (smuellerDD) [inline frontends/server-signal-helper/server_signal_helper.c:61]: Static

**Review by smuellerDD — CHANGES_REQUESTED** [review]
That is a good approach, thanks

**Comment by thillux** [comment]
I should have addressed your comments.

**Comment by smuellerDD** [comment]
Am Montag, 23. Oktober 2023, 14:48:08 CEST schrieb Markus Theil:

Hi Markus,

> I should have addressed your comments.

I had to modify the resume service as I do not have a post-resume.target. Can 
you please have a look at test?

Ciao
Stephan

**Comment by thillux** [comment]
This also works for me :+1:

**Pull request closed without merging**